Python/ceval.c

```diff
@@ -4866,6 +4864,98 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
             DISPATCH();
         }
 
+        TARGET(CALL_NO_KW_LIST_APPEND) {
+            assert(_Py_OPCODE(next_instr[-2]) == PRECALL_METHOD);
+            assert(GET_CACHE()->adaptive.original_oparg == 1);
+            DEOPT_IF(extra_args == 0, CALL_NO_KW);
```

Comment:
Wow, is it possible to not have a bound method? I'm surprised `!PyList_CheckExact(list)` alone isn't enough.

Reply:
Consider `x.append = list; x.append(range(1))`
We would have a stack (tos to the right): `NULL, list, range(1)` and `extra_args == 0`.

Reply:
Yikes, thanks for digging that out.